python/dgl/transform.py

```diff
@@ -463,6 +463,7 @@ def metapath_reachable_graph(g, metapath):
         A homogeneous or bipartite graph.
     """
     adj = 1
+    index_dtype = g._graph.dtype
```

Comment:
It's better to directly use `g.idtype`. Usually we don't rely on graph index any more in those transform functions.